source/SIL.AppBuilder.Portal.Frontend/src/ui/routes/invitations/join-organization/index.tsx

```diff
@@ -0,0 +1,120 @@
+import * as React from 'react';
+import { match as Match, Redirect } from 'react-router';
+import { compose } from 'recompose';
+import { withData, WithDataProps, WithData } from 'react-orbitjs';
+
+import { requireAuth } from '@lib/auth';
+import {patch, tryParseJson} from '@lib/fetch';
+
+import { pathName as notFoundPath } from '@ui/routes/errors/not-found';
+
+import { pushPayload, firstError } from '@data';
+import { withCurrentUserContext, ICurrentUserProps } from '@data/containers/with-current-user';
+
+import OrganizationMembershipInvitiationLoading from './display';
+
+export const pathName = '/invitations/organization-membership/:token';
+
+interface Params {
+  token: string;
+}
+
+interface IOwnProps {
+  match: Match<Params>;
+}
+
+type IProps = IOwnProps & WithDataProps & ICurrentUserProps;
+
+interface IState {
+  isLoading: boolean;
+  error?: Error;
+  organizationMembershipId?: string;
+}
+
+class JoinOrganizationRoute extends React.Component<IProps, IState> {
+  state = {
+    isLoading: true,
+    error: null,
+    organizationMembershipId: null,
+  };
+
+  get token() {
+    return this.props.match.params.token;
+  }
+
+  get hasValidToken(){
+    return this.token && this.token !== '';
+  }
+
+  captureAndThrowError = async (result: any) => {
+    if (result.status === 403 || result.status === 404){
+      let json;
+      try{
+        json = await tryParseJson(result);
+      }
+      catch(error){
+        throw new Error('organization-membership.invite.error.invalid-response');
+      }
+      const error = firstError(json);
+      throw new Error(error.title);
+    }
+    else{
+      const body = await result.text();
+      const err = new Error('organization-membership.invite.error.unexpected');
```

Comment:
thrown errors should be custom error classes in order to reduce error-proneness with the parameters, especially if statically passed

Reply:
I think there are existing errors in `@data/errors`

Reply:
👍